website/content/en/docs/kubebuilder/references/logging.md

```diff
@@ -0,0 +1,266 @@
+
+```Go
+package main
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"  
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var globalLog = logf.Log.WithName("global")
+func main() {
+	// Add the zap logger flag set to the CLI. The flag set must
+	// be added before calling flag.Parse().
+	opts := zap.Options{}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+    
+	logger := zap.New(zap.UseFlagOptions(&opts))
+	logf.SetLogger(logger)
+
+	scopedLog := logf.Log.WithName("scoped")
+
+	globalLog.Info("Printing at INFO level")
+	globalLog.V(1).Info("Printing at DEBUG level")
+	scopedLog.Info("Printing at INFO level")
+	scopedLog.V(1).Info("Printing at DEBUG level")
+}
+```
+
+#### Output using the defaults
+```console
+$ go run main.go
+INFO[0000] Running the operator locally in namespace default. 
+{"level":"info","ts":1587741740.407766,"logger":"global","msg":"Printing at INFO level"}
+{"level":"info","ts":1587741740.407855,"logger":"scoped","msg":"Printing at INFO level"}
+```
+
+#### Output overriding the log level to 1 (debug)
+```console
+$ go run main.go --zap-log-level=debug
+INFO[0000] Running the operator locally in namespace default. 
+{"level":"info","ts":1587741837.602911,"logger":"global","msg":"Printing at INFO level"}
+{"level":"debug","ts":1587741837.602964,"logger":"global","msg":"Printing at DEBUG level"}
+{"level":"info","ts":1587741837.6029708,"logger":"scoped","msg":"Printing at INFO level"}
+{"level":"debug","ts":1587741837.602973,"logger":"scoped","msg":"Printing at DEBUG level"}
+```
+## Custom zap logger
+
+In order to use a custom zap logger, [`zap`][controller_runtime_zap] from controller-runtime can be utilized to wrap it in a `logr` implementation.
+
+Below is an example illustrating the use of [`zap-logfmt`][logfmt_repo] in logging.
+
+### Example
+
+In your `main.go` file, replace the current implementation for logs inside the `main` function:
+
+```Go
+...
+// Add the zap logger flag set to the CLI. The flag set must
+// be added before calling flag.Parse().
+	opts := zap.Options{}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	logger := zap.New(zap.UseFlagOptions(&opts))
+	logf.SetLogger(logger)
+...
+```
+
+With:
+
+```Go
+	configLog := zapcr.NewProductionEncoderConfig()
+	configLog.EncodeTime = func(ts time.Time, encoder zapcore.PrimitiveArrayEncoder) {
+		encoder.AppendString(ts.UTC().Format(time.RFC3339Nano))
+	}
+	logfmtEncoder := zaplogfmt.NewEncoder(configLog)
+
+	// Construct a new logr.logger.
+	logger := zap.New(zap.UseDevMode(true), zap.WriteTo(os.Stdout), zap.Encoder(logfmtEncoder))
+
+	// Set the controller logger to log, which will
+	// be propagated through the whole operator, generating
+	// uniform and structured logs.
+	logf.SetLogger(logger)
+```
+
+Ensure that the following additional imports are being used:
+
+```Go
+import(
+	...
+	zaplogfmt "github.com/sykesm/zap-logfmt"
+	zapcr "go.uber.org/zap"
```

Comment:
Why is this aliased as `zapcr`? I almost read it as zap-controller-runtime but it's actually the upstream zap pkg.
Maybe rename this as `uzap`. Or alias the controller-runtime one. `zapcr "sigs.k8s.io/controller-runtime/pkg/log/zap"`.

Reply:
@bharathi-tenneti Can we also move these imports up into the above snippet? It's hard to read that example without the imports first. Just make it into one complete example block.